### PR TITLE
Eobs 418 patient refusal news refuse observation process

### PR DESCRIPTION
**Pull request by moderatemisbehaviour**
* New logic for partial observation with reason _refused_.
* Instead of creating an new observation due at the same time as the refused observation was originally due, the due time will be calculated from the moment of refusal based on the patient's current clinical risk. The behaviour is closer to normal observations but with the frequency capped at 24 hours.
* If a patient refuses observation straight after a patient monitoring exception a new obs will be created due in 4 hours time.

### CODE REVIEW

File: nh_ews/ews.py [hunk 1]
```diff
@@ -782,6 +924,7 @@ def create_activity(self, cr, uid, vals_activity=None, vals_data=None,
             nh_clinical_patient_observation_ews, self).create_activity(
             cr, uid, vals_activity, vals_data, context=context)
 
+    # TODO Make generic and move to nh.clinical.patient.observation model.
```
Comment (moderatemisbehaviour) [inline nh_ews/ews.py]: Use new tech debt process for this TODO.

**Comment by Gimpneek** [comment]
:shipit: Once TODO's sorted

**Comment by Gimpneek** [comment]
:shipit:

File: nh_observations/frequencies.py [hunk 2]
```diff
@@ -1,20 +1,71 @@
...
+EVERY_3_DAYS = (THREE_DAYS, 'Every 3 Days')
+EVERY_WEEK = (ONE_WEEK, 'Every Week')
+
+# To summarise the mappings below, frequencies after refusals are capped at
+# 24 hours.
+#
+# Could have just made a simple function but decided to write them out long
+# hand as they may change in the future. Return the tuple with the label rather
+# than just the time so it can be useful in more scenarios.
+PATIENT_REFUSAL_ADJUSTMENTS = {
```
Comment (Gimpneek) [inline nh_observations/frequencies.py]: This has the potential to cause issues. Where you're updating the first value in the frequency tuples you need to ensure this isn't updating the original object itself and instead returning a copy

Comment (Gimpneek) [inline nh_observations/frequencies.py]: Ah actually looking at it again (I originally looked at this due to the issues Nayi was having yesterday) you're just using the first item in the tuple for the key in the dictionary. I think I was under the impression at the time that this was reassigning the value. 

In that case no cause for concern!

Comment (moderatemisbehaviour) [inline nh_observations/frequencies.py]: Better safe than sorry! Cheers 👍

File: nh_eobs/tests/common/test_data_creator.py [hunk 3]
```diff
@@ -5,15 +5,17 @@ def admit_and_place_patient(self):
 
 
 def admit_patient(self):
-    self.api_model = self.env['nh.eobs.api']
+    try:
+        self.api_model = self.env['nh.eobs.api']
+    except KeyError:
+        self.api_model = self.env['nh.clinical.api']
```
Comment (Gimpneek) [inline nh_eobs/tests/common/test_data_creator.py]: Why not just use `nh.clinical.api` regardless?

File: nh_eobs/tests/test_get_last_full_obs_activity.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    def test_returns_obs_from_before_transfer_after_placement(
+            self):
+        self.initial_medium_risk_obs_activity = \
+            self.observation_test_utils.create_ews_obs_activity(
+                self.patient.id, self.spell_activity_id,
+                clinical_risk_sample_data.MEDIUM_RISK_DATA
+            )
+
+        self.api_model.transfer(self.hospital_number, {'location': 'WB'})
```
Comment (Gimpneek) [inline nh_eobs/tests/test_get_last_full_obs_activity.py]: Does the `test_data_creator.admit_and_place_patient()` method create the location with code 'WB'?

File: nh_observations/tests/nh_clinical_patient_observation/test_observations.py [hunk 5]
```diff
@@ -1,9 +1,10 @@
 # -*- coding: utf-8 -*-
 # Part of Open eObs. See LICENSE file for full copyright and licensing details.
-from openerp.tests import common
-from faker import Faker
 import logging
 
+from faker import Faker
```
Comment (Gimpneek) [inline nh_observations/tests/nh_clinical_patient_observation/test_observations.py]: This test will most likely need refactoring if it's using faker (these tests will occassionally fail due to clashes with existing data)

File: nh_eobs/tests/common/test_data_creator.py (hunk 3 above)
Comment (moderatemisbehaviour) [inline nh_eobs/tests/common/test_data_creator.py]: I wanted to be able to use it from multiple different levels so if `nh.eobs.api` was available I figured we should use it so we don't skip logic that might affect the outcome of the test.

File: nh_eobs/tests/test_get_last_full_obs_activity.py (hunk 4 above)
Comment (moderatemisbehaviour) [inline nh_eobs/tests/test_get_last_full_obs_activity.py]: Yeah it creates 2 wards, 'WA' and 'WB'.

File: nh_eobs/tests/common/test_data_creator.py (hunk 3 above)
Comment (Gimpneek) [inline nh_eobs/tests/common/test_data_creator.py]: Ok but this test is in nh_eobs which contains the nh.eobs.api model so there should never be an instance where nh.eobs.api isn't available for this test

**Comment by Gimpneek** [comment]
:shipit:

**Comment by Nayira** [comment]
Test for story EOBS-418 and feature branch bugs EOBS-706, EOBS-709, EOBS-714, EOBS-715 and EOBS-716 have Passed. :shipit: !

**Comment by Gimpneek** [comment]
unit test :shipit:

**Pull request merged**